Disallow sync GeoServer layers for groups where user is not a member

ricardogsilva opened the pull request:
This PR checks that user is a member of the current group before allowing the call to sync GeoServer layers

Fixes #101

lucernae commented:
Ok, I think I misunderstand the original requirements.

Does that mean you want to check the "user's" permission?
Wouldn't that mean we should check in the template as well?

lucernae commented on geonode_dominode/geonode_dominode/templates/groups/group_detail_override.html:
```diff
@@ -55,15 +55,15 @@ <h2 class="page-title">{{ object.title|default:object.slug }}</h2>
-          <input type="submit" class="btn btn-primary" value="{% trans "Sync GeoServer" %}" />
-        </form>
-      </li>
+      {% if is_member and perms.groups.can_sync_geoserver %}
```
Now I am unsure `perms.groups` is what we really want. Because based on the issue that you raised, that means we want to check the perms of the group that is currently being viewed?

lucernae replied:
Check the attached PR in your Repo, I made a small diff.

ricardogsilva commented:
@lucernae 

I checked your PR on my repo, and thought a bit about it. However, I want to go a different way with this.

You are right in spotting that this is not a robust solution just yet. I think what we need is to leverage [django-guardian], which GeoNode alreay uses, to assign permissions on each group object, not on the whole `Group` model.

The plan is to add some code that listens to the GroupProfile's `post_save()` signal and assigns the permission to each group object that matches a criteria (I don't want all groups to get this permission - for example, if we add a `cors` group I don't want this group to be able to sync GeoServer layers). 

Since we are not in control of the `GroupProfile` model (comes from upstream GeoNode), the initial criteria for deciding if the group shall be given the permission is if its name is of the form `{somename-editor}`.

I'll work on the changes later on and then I'll request your review


[django-guardian]: https://django-guardian.readthedocs.io/en/stable/userguide/check.html

ricardogsilva commented:
@lucernae 

PR is now ready for antoher review. Recap of implementation:

- `dajngo-guardian` is now used to assign object-level permissions to each existing group. This means the `groups.can_syn_geoserver` permission is now added to each individual group - group members are now able to sync their own groups, but not other groups

- the permissions are now granted to groups that are given the `dominode-editor` category. Previously we were doing a name match to be able to see if the permission should be added. The permission is added by a signal handler that is listening for the `m2m_changed` signal
 
- changed the template to only show the geoserver sync button in case the user has permission to modify the current `GroupProfile` (which he will only have if he is a member of the group)

-  changed the view to check for the relevant permission too

lucernae commented on geonode_dominode/geonode_dominode/views.py:
```diff
@@ -1,60 +1,49 @@
+    GEOSERVER_SYNC_PERM_CODE,
+)
 
+DEPARTMENT_GROUP_PROFILE_SUFFIX = '-editor'
```
Why not move this to `.constants` too?

ricardogsilva replied:
because it is only used in this module.

lucernae reviewed (commented):
lgtm!
